documentation/asciidoc/computers/linux_kernel/building.adoc

```diff
@@ -1,342 +1,462 @@
++
+[source,console]
 ----
-make -j4 Image.gz modules dtbs
-sudo make modules_install
-sudo cp arch/arm64/boot/dts/broadcom/*.dtb /boot/firmware/
-sudo cp arch/arm64/boot/dts/overlays/*.dtb* /boot/firmware/overlays/
-sudo cp arch/arm64/boot/dts/overlays/README /boot/firmware/overlays/
-sudo cp arch/arm64/boot/Image.gz /boot/firmware/$KERNEL.img
+$ make -j4 zImage modules dtbs
+$ sudo make modules_install
 ----
 
-NOTE: On Raspberry Pi 2, 3, 4 and 5, the `-j4` flag splits the work between all four cores, speeding up compilation significantly.
+. Depending on your kernel version, run the following command:
+  * For kernels up to version 6.4:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/*.dtb /boot/firmware/
+----
+* For kernels version 6.5 and above:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/broadcom/*.dtb /boot/firmware/
+----
+. Finally, copy over the overlays, README, and image:
++
+[source,console]
+----
+$ sudo cp arch/arm/boot/dts/overlays/*.dtb* /boot/firmware/overlays/
+$ sudo cp arch/arm/boot/dts/overlays/README /boot/firmware/overlays/
+$ sudo cp arch/arm/boot/zImage /boot/firmware/$KERNEL.img
+----
 
-If you now reboot, your Raspberry Pi should be running your freshly-compiled kernel.
+TIP: On multi-core Raspberry Pi models, the `-j <n>` option distributes work between cores. This can speed up compilation significantly. Run `nproc` to see how many processors you have; we recommend passing a number 1.5x your number of processors.
 
-=== Cross-compiling the kernel
+Run the following command to reboot your Raspberry Pi and run your freshly-compiled kernel:
 
-First, you will need a suitable Linux cross-compilation host. We tend to use Ubuntu; since Raspberry Pi OS is also a Debian distribution, it means many aspects are similar, such as the command lines.
+[source,console]
+----
+$ sudo reboot
+----
 
-You can either do this using VirtualBox (or VMWare) on Windows, or install it directly onto your computer. 
+=== Cross-compiled builds
+
+First, you will need a suitable Linux cross-compilation host. We tend to use Ubuntu; since Raspberry Pi OS is also a Debian distribution, compilation commands are similar.
 
 ==== Install required dependencies and toolchain
 
-To build the sources for cross-compilation, make sure you have the dependencies needed on your machine by executing:
+To build the sources for cross-compilation, install the required dependencies onto your device. Run the following command to install most dependencies:
 
-[,bash]
+[source,console]
 ----
-sudo apt install git bc bison flex libssl-dev make libc6-dev libncurses5-dev
+$ sudo apt install git bc bison flex libssl-dev make libc6-dev libncurses5-dev
 ----
 
-===== Install the 32-bit toolchain for a 32-bit kernel
+Then, install the proper toolchain for the kernel architecture you wish to build:
+
+* To install the 64-bit toolchain to build a 64-bit kernel, run the following command:
```

Comment:
Again, feels slightly strange to mention 64-bit before 32-bit? :shrug:

Reply:
With Pi 5 dropping 32-bit support entirely (I think?), I think it's time we embraced 64-bit as the default, with 32-bit delegated to a 'legacy support' kind of position. Especially with the 64-bit kernel stuff on a few other devices.